editor/plugins/path_3d_editor_plugin.cpp

```diff
@@ -436,8 +443,42 @@ void Path3DGizmo::redraw() {
 			add_lines(tilt_handle_lines, path_tilt_material);
 		}
 
-		if (primary_handle_points.size()) {
-			add_handles(primary_handle_points, handles_material);
+		// Primary handles: manage different colors for first, intermediate and last points.
+		const int pc = primary_handle_points.size();
+		if (pc) {
+			// Need to define indices separately.
+			Vector<int> idx;
+			idx.resize(pc);
+			for (int j = 0; j < pc; j++) {
+				idx.write[j] = j;
```

Comment:
```suggestion
			int *idx_ptr = idx.ptrw();
			for (int j = 0; j < pc; j++) {
				idx_ptr[j] = j;
```
More efficient version, avoids COW